packages/path_provider/path_provider/README.md

```diff
@@ -2,28 +2,206 @@
+
+### External Storage Directory
+Path to a directory where the application may access top-level storage.
+This path is typically an SD card or built-in storage partition in a device. 
+The current operating system should be determined before issuing this
+function call, as this functionality is **only available on Android**.
+
+On other than Android platform, this function throws an `UnsupportedError` as no equivalent
+path exists.
+
+Example:
+``` dart
+Future<List<Directory>?> requestDirectories() async {
+  try {
+    final Directory? directory = await getExternalStorageDirectory();
+    final String? dirPath = directory?.path;
+    return dirPath;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
 
-### Usage in tests
 
-`path_provider` now uses a `PlatformInterface`, meaning that not all platforms share the a single `PlatformChannel`-based implementation.
+### External Cache Directories
+Paths to directories where application-specific external cache data can be
+stored. These paths typically reside on external storage like separate
+partitions or SD cards. Phones may have multiple storage directories
+available.
+
+The current operating system should be determined before issuing this
+function call, as this functionality is **only available on Android**.
+
+On other than Android platform, this function throws an `UnsupportedError` as no equivalent
+path exists.
+
+Example:
+``` dart
+Future<List<Directory>?> requestDirectories() async {
+  try {
+    final List<Directory>? directories = await getExternalCacheDirectories();
+    return directories;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
+
+
+### External Storage Directories
+Paths to directories where application-specific data can be stored.
+These paths typically reside on external storage like separate partitions
+or SD cards. Phones may have multiple storage directories available.
+
+The current operating system should be determined before issuing this
+function call, as this functionality is **only available on Android**.
+
+On other than Android platform, this function throws an `UnsupportedError` as no equivalent
+path exists.
+
+Example:
+``` dart
+Future<List<Directory>?> requestDirectories() async {
+  try {
+    final List<Directory>? directories = 
+        await getExternalStorageDirectories(type: StorageDirectory.pictures);
+    return directories;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
+
+### Downloads Directory
+Path to the directory where downloaded files can be stored.
+This is typically only relevant on desktop operating systems.
+
+On Android and iOS, this function throws an `UnsupportedError` as no equivalent
+path exists.
+
+Example:
+``` dart
+Future<String?> requestDirectoryPath() async {
+  try {
+    final Directory? directory = await getDownloadsDirectory();
+    final String? dirPath = directory?.path;
+    return dirPath;
+  } catch (e) {
+    // HANDLE ERROR
+  }
+  return null;
+}
+```
```

Comment:
I think we should remove all of these sections. They don't seem to add any value over the API docs, and the examples are entirely rote.

If we want an example, we could show a single mini-example that shows a few common examples (support dir, documents directory, temp).

Reply:
dropped out boilerplate.